### ISSUE TITLE
Error when creating assignment

### ISSUE THREAD

**Issue by whositwhatnow** [jupyter/nbgrader#556]
Finally figured out how to get this going but am running into an issue when creating my first assignment.
I copy assignment i created into the correct directory.  when i run nbgrader assign assignement i get hte following error.  I have zero clue what im loking for here 

[root@jupyterhub lab1]# nbgrader assign ps1
[AssignApp | INFO] Updating/creating assignment 'ps1': {}
[AssignApp | INFO] Converting notebook /var/courses/BIO362Assignments/lab1/source/./ps1/Exercise 1 - DNA is a string.ipynb to notebook
[AssignApp | ERROR] There was an error processing assignment: /var/courses/BIO362Assignments/lab1/source/./ps1
[AssignApp | ERROR] Traceback (most recent call last):
      File "/usr/local/lib/python3.5/site-packages/nbgrader/api.py", line 1615, in find_source_cell
        .filter(SourceCell.name == name, Notebook.name == notebook, Assignment.name == assignment)\
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/query.py", line 2724, in one
        raise orm_exc.NoResultFound("No row was found for one()")
    sqlalchemy.orm.exc.NoResultFound: No row was found for one()

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/usr/local/lib/python3.5/site-packages/nbgrader/api.py", line 1644, in update_or_create_source_cell
        source_cell = self.find_source_cell(name, notebook, assignment)
      File "/usr/local/lib/python3.5/site-packages/nbgrader/api.py", line 1618, in find_source_cell
        raise MissingEntry("No such source cell: {}/{}/{}".format(assignment, notebook, name))
    nbgrader.api.MissingEntry: No such source cell: ps1/Exercise 1 - DNA is a string/Read_16

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
        context)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
        cursor.execute(statement, parameters)
    sqlite3.IntegrityError: constraint failed

    The above exception was the direct cause of the following exception:

    Traceback (most recent call last):
      File "/usr/local/lib/python3.5/site-packages/nbgrader/api.py", line 1587, in add_source_cell
        self.db.commit()
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/scoping.py", line 157, in do
        return getattr(self.registry(), name)(*args, **kwargs)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/session.py", line 801, in commit
        self.transaction.commit()
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/session.py", line 392, in commit
        self._prepare_impl()
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/session.py", line 372, in _prepare_impl
        self.session.flush()
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/session.py", line 2019, in flush
        self._flush(objects)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/session.py", line 2137, in _flush
        transaction.rollback(_capture_exception=True)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/util/langhelpers.py", line 60, in __exit__
        compat.reraise(exc_type, exc_value, exc_tb)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/util/compat.py", line 186, in reraise
        raise value
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/session.py", line 2101, in _flush
        flush_context.execute()
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/unitofwork.py", line 373, in execute
        rec.execute(self)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/unitofwork.py", line 532, in execute
        uow
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/persistence.py", line 174, in save_obj
        mapper, table, insert)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/orm/persistence.py", line 800, in _emit_insert_statements
        execute(statement, params)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 914, in execute
        return meth(self, multiparams, params)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/sql/elements.py", line 323, in _execute_on_connection
        return connection._execute_clauseelement(self, multiparams, params)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1010, in _execute_clauseelement
        compiled_sql, distilled_params
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
        context)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1341, in _handle_dbapi_exception
        exc_info
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/util/compat.py", line 202, in raise_from_cause
        reraise(type(exception), exception, tb=exc_tb, cause=cause)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/util/compat.py", line 185, in reraise
        raise value.with_traceback(tb)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
        context)
      File "/usr/local/lib/python3.5/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute
        cursor.execute(statement, parameters)
    sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) constraint failed [SQL: 'INSERT INTO source_cell (id, name, cell_type, locked, source, checksum, notebook_id) VALUES (?, ?, ?, ?, ?, ?, ?)'] [parameters: ('4f5fac8bc34d47cb93c612ef1da8f856', 'Read_16', 'raw', 1, "print('My DNA sequence is: ' + my_dna)\n", '31dcbc5fe3530d6aad05303467aaa59c', '929dc3e14f61400c88b7eb02297a0e79')]

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/usr/local/lib/python3.5/site-packages/nbgrader/apps/baseapp.py", line 769, in convert_notebooks
        super(BaseNbConvertApp, self).convert_notebooks()
      File "/usr/local/lib/python3.5/site-packages/nbconvert/nbconvertapp.py", line 457, in convert_notebooks
        self.convert_single_notebook(notebook_filename)
      File "/usr/local/lib/python3.5/site-packages/nbconvert/nbconvertapp.py", line 428, in convert_single_notebook
        output, resources = self.export_single_notebook(notebook_filename, resources, input_buffer=input_buffer)
      File "/usr/local/lib/python3.5/site-packages/nbconvert/nbconvertapp.py", line 357, in export_single_notebook
        output, resources = self.exporter.from_filename(notebook_filename, resources=resources)
      File "/usr/local/lib/python3.5/site-packages/nbconvert/exporters/exporter.py", line 165, in from_filename
        return self.from_file(f, resources=resources, **kw)
      File "/usr/local/lib/python3.5/site-packages/nbconvert/exporters/exporter.py", line 183, in from_file
        return self.from_notebook_node(nbformat.read(file_stream, as_version=4), resources=resources, **kw)
      File "/usr/local/lib/python3.5/site-packages/nbconvert/exporters/notebook.py", line 26, in from_notebook_node
        nb_copy, resources = super(NotebookExporter, self).from_notebook_node(nb, resources, **kw)
      File "/usr/local/lib/python3.5/site-packages/nbconvert/exporters/exporter.py", line 130, in from_notebook_node
        nb_copy, resources = self._preprocess(nb_copy, resources)
      File "/usr/local/lib/python3.5/site-packages/nbconvert/exporters/exporter.py", line 302, in _preprocess
        nbc, resc = preprocessor(nbc, resc)
      File "/usr/local/lib/python3.5/site-packages/nbconvert/preprocessors/base.py", line 47, in __call__
        return self.preprocess(nb,resources)
      File "/usr/local/lib/python3.5/site-packages/nbgrader/preprocessors/savecells.py", line 82, in preprocess
        self._create_notebook()
      File "/usr/local/lib/python3.5/site-packages/nbgrader/preprocessors/savecells.py", line 56, in _create_notebook
        source_cell = self.gradebook.update_or_create_source_cell(name, self.notebook_id, self.assignment_id, **info)
      File "/usr/local/lib/python3.5/site-packages/nbgrader/api.py", line 1646, in update_or_create_source_cell
        source_cell = self.add_source_cell(name, notebook, assignment, **kwargs)
      File "/usr/local/lib/python3.5/site-packages/nbgrader/api.py", line 1590, in add_source_cell
        raise InvalidEntry(*e.args)
    nbgrader.api.InvalidEntry: (sqlite3.IntegrityError) constraint failed

[AssignApp | ERROR] There was an error processing assignment 'ps1' for student '.'
[AssignApp | ERROR] Please see the error log (.nbgrader.log) for details on the specific errors on the above failures.

**Comment by jhamrick** [comment]
Hmm, it's hard to know exactly what is going wrong without seeing the notebook. Would you mind posting it somewhere that I can look at the notebook or emailing it to me?

**Comment by whositwhatnow** [comment]
I found the offending cell and removed it for now.  it runs correctly and student version is created.  May have been a syntax error in my cell.

One thing i did notice is that my answer cells are showing up in the assignment.

I'll email a copy.

**Comment by jhamrick** [comment]
The answer cell that is showing up seems to be marked as autograder tests, rather than an autograded answer (and the test cell below it is marked as an autograded answer, rather than as autograder tests).

**Comment by whositwhatnow** [comment]
Is there a way to hide autograder test from the students?

**Comment by jhamrick** [comment]
No, unfortunately, not at the moment: http://nbgrader.readthedocs.io/en/stable/user_guide/faq.html#can-i-hide-the-test-cells-in-a-nbgrader-assignment

**Comment by whositwhatnow** [comment]
syntax error in my cell.  My apologies, i'm a total newbie :(

**Comment by jhamrick** [comment]
No worries!

I'm a bit concerned though that a syntax error generated that long traceback -- that's something that should provide a better error message at the very least. Can you let me know what the syntax error was?

**Comment by Szepi** [comment]
Hmm, I am getting the same error.
In fact, I tried:
```bash
$ nbgrader quickstart MyCourse
$ cd MyCourse
$ nbgrader assign ps1
```
and I get the same error. Earlier, everything was working fine.
Below is a copy of my log file.
[nbgrader.log.txt](https://github.com/jupyter/nbgrader/files/892178/nbgrader.log.txt)
I am using nbgrader 0.5.0
I am thinking that somehow I messed up with the nbextension or serverextension (at some point I started manually changing config files for these to get FormGrader working).
